src/main/scala/better/files/package.scala

```diff
@@ -395,6 +397,79 @@ package object files {
     def buffered: BufferedWriter = new BufferedWriter(writer)
   }
 
+  implicit class ZipOps(basePath: File) {
+
+    def zip: Try[File] = {
+      val zipFile = File.newTemp(basePath.nameWithoutExtension, ".zip")
+
+      val zipEntries = basePath match {
+        case directory if directory.isDirectory =>
+          directory.listRecursively()
+            .filter(_.fullPath != directory.fullPath)
+            .collect {
+            case file if file.isDirectory =>
+              val relativeDirPath = directory.path.relativize(file.path).toString
+              // make sure it ends with / for ZipEntry to consider it a directory
+              val cleanRelativeDirPath = relativeDirPath.stripPrefix("/").stripSuffix("/")
+              val zipRelativeDirPath = s"$cleanRelativeDirPath/"
+              (file, new ZipEntry(zipRelativeDirPath))
+
+            case file =>
+              (file, new ZipEntry(directory.path.relativize(file.path).toString))
+          }
+
+        case file =>
+          val relativePath = file.parent.path.relativize(file.path).toString
+          List((file, new ZipEntry(relativePath)))
+      }
+
+      Try {
+        val zipOutputStream = new ZipOutputStream(zipFile.out)
+        zipEntries.foreach {
+          case (file, entry) if file.isDirectory =>
+            zipOutputStream.putNextEntry(entry)
+            zipOutputStream.closeEntry()
+
+          case (file, entry) =>
+            zipOutputStream.putNextEntry(entry)
+            zipOutputStream.write(file.bytes.toArray)
+            zipOutputStream.closeEntry()
+        }
+
+        zipOutputStream.close()
+
+        zipFile
+      }
+    }
+
+    def unzip: Try[File] = {
+      val outputDir = File.newTempDir(basePath.nameWithoutExtension)
+
+      Try {
+        val zipArchive = new ZipFile(basePath.toJava)
+        val (directories, files) = enumerationAsScalaIterator(zipArchive.entries()).partition(_.isDirectory)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/pathikrit/better-files/file/1014773478/issues/source?bid=2860326&fileBranchId=2875696#l450)